lib/actual_db_schema/schema_diff.rb

```diff
@@ -0,0 +1,204 @@
+        next unless (md = line_content.match(regex))
+
+        action_proc = action_map[kind]
+        return action_proc.call(md) if action_proc
+      end
+
+      [nil, nil]
+    end
+
+    def guess_action(sign, table, col_name)
+      case sign
+      when "+"
+        old_table = parsed_old_schema[table] || {}
+        old_table[col_name].nil? ? :add_column : :change_column
+      when "-"
+        new_table = parsed_new_schema[table] || {}
+        new_table[col_name].nil? ? :remove_column : :change_column
+      end
+    end
+
+    def find_table_in_new_schema(new_line_number)
+      current_table = nil
+
+      new_schema_content.lines[0...new_line_number].each do |line|
+        if (match = line.match(/create_table\s+["']([^"']+)["']/))
+          current_table = match[1]
+        end
+      end
+      current_table
+    end
+
+    def find_migrations(action, table_name, col_or_index_name)
+      matches = []
+
+      migration_changes.each do |version, changes|
+        changes.each do |chg|
+          next unless chg[:table].to_s == table_name.to_s
+
+          matches << version if migration_matches?(chg, action, col_or_index_name)
+        end
+      end
+
+      matches
+    end
+
+    def migration_matches?(chg, action, col_or_index_name)
+      return (chg[:action] == action) if col_or_index_name.nil?
+
+      matchers = {
+        rename_column: -> { rename_column_matches?(chg, action, col_or_index_name) },
+        rename_index: -> { rename_index_matches?(chg, action, col_or_index_name) },
+        add_index: -> { index_matches?(chg, action, col_or_index_name) },
+        remove_index: -> { index_matches?(chg, action, col_or_index_name) }
+      }
+
+      matchers.fetch(chg[:action], -> { column_matches?(chg, action, col_or_index_name) }).call
+    end
+
+    def rename_column_matches?(chg, action, col)
+      (action == :remove_column && chg[:old_column].to_s == col.to_s) ||
+        (action == :add_column && chg[:new_column].to_s == col.to_s)
+    end
+
+    def rename_index_matches?(chg, action, name)
+      (action == :remove_index && chg[:old_name] == name) ||
+        (action == :add_index && chg[:new_name] == name)
+    end
+
+    def index_matches?(chg, action, col_or_index_name)
+      return false unless chg[:action] == action
+
+      extract_migration_index_name(chg, chg[:table]) == col_or_index_name.to_s
+    end
+
+    def column_matches?(chg, action, col_name)
+      chg[:column] && chg[:column].to_s == col_name.to_s && chg[:action] == action
+    end
+
+    def extract_migration_index_name(chg, table_name)
+      return chg[:options][:name].to_s if chg[:options].is_a?(Hash) && chg[:options][:name]
+
+      return "" unless (columns = chg[:columns])
+
+      cols = columns.is_a?(Array) ? columns : [columns]
+      "index_#{table_name}_on_#{cols.join("_and_")}"
+    end
+
+    def annotate_line(line, migration_versions)
+      "#{line.chomp}#{colorize(" // #{migration_versions.join(", ")} //", :gray)}\n"
```

Comment:
Let's show the relative file path instead. This makes it easier to open the file, as some terminals (such as iTerm) allow you to open files by clicking on the file path.